# Import dataset

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('cbb.csv')

# Inspect dataset

In [3]:
## view the first 5 rows of the dataset
df.head()

TEAM CONF   G   W  ADJOE  ADJDE  BARTHAG  EFG_O  EFG_D   TOR  \
0  North Carolina  ACC  40  33  123.3   94.9   0.9531   52.6   48.1  15.4   
1       Wisconsin  B10  40  36  129.1   93.6   0.9758   54.8   47.7  12.4   
2        Michigan  B10  40  33  114.4   90.4   0.9375   53.9   47.7  14.0   
3      Texas Tech  B12  38  31  115.2   85.2   0.9696   53.5   43.0  17.7   
4         Gonzaga  WCC  39  37  117.8   86.3   0.9728   56.6   41.1  16.2   

   ...  FTRD  2P_O  2P_D  3P_O  3P_D  ADJ_T   WAB  POSTSEASON  SEED  YEAR  
0  ...  30.4  53.9  44.6  32.7  36.2   71.7   8.6         2ND   1.0  2016  
1  ...  22.4  54.8  44.7  36.5  37.5   59.3  11.3         2ND   1.0  2015  
2  ...  30.0  54.7  46.8  35.2  33.2   65.9   6.9         2ND   3.0  2018  
3  ...  36.6  52.8  41.9  36.5  29.7   67.5   7.0         2ND   3.0  2019  
4  ...  26.9  56.3  40.0  38.2  29.0   71.5   7.7         2ND   1.0  2017  

[5 rows x 24 columns]

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1757 entries, 0 to 1756
Data columns (total 24 columns):
TEAM          1757 non-null object
CONF          1757 non-null object
G             1757 non-null int64
W             1757 non-null int64
ADJOE         1757 non-null float64
ADJDE         1757 non-null float64
BARTHAG       1757 non-null float64
EFG_O         1757 non-null float64
EFG_D         1757 non-null float64
TOR           1757 non-null float64
TORD          1757 non-null float64
ORB           1757 non-null float64
DRB           1757 non-null float64
FTR           1757 non-null float64
FTRD          1757 non-null float64
2P_O          1757 non-null float64
2P_D          1757 non-null float64
3P_O          1757 non-null float64
3P_D          1757 non-null float64
ADJ_T         1757 non-null float64
WAB           1757 non-null float64
POSTSEASON    340 non-null object
SEED          340 non-null float64
YEAR          1757 non-null int64
dtypes: float64(18), int64(3), object(3)

In [5]:
df.describe()

G            W        ADJOE        ADJDE      BARTHAG  \
count  1757.000000  1757.000000  1757.000000  1757.000000  1757.000000   
mean     31.523051    16.513375   103.542402   103.542459     0.493398   
std       2.602819     6.545571     7.304975     6.472676     0.255291   
min      24.000000     0.000000    76.700000    84.000000     0.007700   
25%      30.000000    12.000000    98.600000    98.900000     0.283700   
50%      31.000000    16.000000   103.100000   103.800000     0.474000   
75%      33.000000    21.000000   108.100000   108.000000     0.710600   
max      40.000000    38.000000   129.100000   124.000000     0.984200   

             EFG_O        EFG_D          TOR         TORD          ORB  ...  \
count  1757.000000  1757.000000  1757.000000  1757.000000  1757.000000  ...   
mean     50.120489    50.312806    18.591804    18.521286    29.277120  ...   
std       3.130430     2.859604     1.991637     2.108968     4.101782  ...   
min      39.400000    39.600000    12.400000    10.200000    15.000000  ...   
25%      48.100000    48.400000    17.200000    17.100000    26.600000  ...   
50%      50.000000    50.300000    18.500000    18.500000    29.400000  ...   
75%      52.100000    52.300000    19.800000    19.900000    31.900000  ...   
max      59.800000    59.500000    26.100000    28.000000    42.100000  ...   

               FTR         FTRD         2P_O         2P_D         3P_O  \
count  1757.000000  1757.000000  1757.000000  1757.000000  1757.000000   
mean     35.097894    35.373307    49.135970    49.298065    34.563517   
std       4.884599     5.900935     3.422136     3.288265     2.742323   
min      21.600000    21.800000    37.700000    37.700000    25.200000   
25%      31.700000    31.200000    46.900000    47.100000    32.600000   
50%      34.900000    34.900000    49.000000    49.300000    34.600000   
75%      38.300000    39.200000    51.400000    51.500000    36.400000   
max      51.000000    58.500000    62.600000    61.200000    44.100000   

              3P_D        ADJ_T          WAB        SEED         YEAR  
count  1757.000000  1757.000000  1757.000000  340.000000  1757.000000  
mean     34.744792    68.422254    -7.837109    8.791176  2017.002277  
std       2.369727     3.258920     6.988694    4.674090     1.415419  
min      27.100000    57.200000   -25.200000    1.000000  2015.000000  
25%      33.100000    66.400000   -13.000000    5.000000  2016.000000  
50%      34.700000    68.500000    -8.400000    9.000000  2017.000000  
75%      36.300000    70.400000    -3.100000   13.000000  2018.000000  
max      43.100000    83.400000    13.100000   16.000000  2019.000000  

[8 rows x 21 columns]

In [6]:
## size of dataset
df.shape

(1757, 24)

In [7]:
## any NA values
df.isnull().sum()

TEAM             0
CONF             0
G                0
W                0
ADJOE            0
ADJDE            0
BARTHAG          0
EFG_O            0
EFG_D            0
TOR              0
TORD             0
ORB              0
DRB              0
FTR              0
FTRD             0
2P_O             0
2P_D             0
3P_O             0
3P_D             0
ADJ_T            0
WAB              0
POSTSEASON    1417
SEED          1417
YEAR             0
dtype: int64

#### From this, I can conclude/confirm that there are only two columns that contain NA values. In the case of this dataset and for the purposes of this project, these NA values actually tell us something: that "(1757-340)= 1417" teams have never made it to the March Madness tournament.

### Inspecting the target column, "SEED"

In [8]:
df['SEED'].values

array([ 1.,  1.,  3., ...,  2., 11.,  4.])

In [9]:
df['SEED'].value_counts

<bound method IndexOpsMixin.value_counts of 0        1.0
1        1.0
2        3.0
3        3.0
4        1.0
        ... 
1752     7.0
1753     3.0
1754     2.0
1755    11.0
1756     4.0
Name: SEED, Length: 1757, dtype: float64>

In [10]:
print(np.min(df['SEED']))
print(np.max(df['SEED']))

1.0
16.0


#### This informs me that the highest SEED number is 16, for each basketball tournament. (Refer to *00_dataset-variables.ipynb* for more information on "SEED")

In [11]:
df['SEED'].fillna(value=0, inplace = True)

In [12]:
df['SEED'].isnull().sum()

0

#### The null values in the dataset have been converted to 0s. 

In [13]:
df['YEAR'].unique()

array([2016, 2015, 2018, 2019, 2017])

#### The dataset spans 5 years.